apps/desktop-wallet/src/pages/UnlockedWallet/OverviewPage/AmountsOverviewPanel.tsx

```diff
@@ -69,9 +70,9 @@ const AmountsOverviewPanel: FC<AmountsOverviewPanelProps> = ({ className, addres
   const selectAddressesHaveHistoricBalances = useMemo(makeSelectAddressesHaveHistoricBalances, [])
   const hasHistoricBalances = useAppSelector((s) => selectAddressesHaveHistoricBalances(s, addressHashes))
   const fiatCurrency = useAppSelector((s) => s.settings.fiatCurrency)
-  const { data: price, isLoading: isPriceLoading } = useGetPriceQuery(currencies[fiatCurrency].ticker, {
-    pollingInterval: 60000
-  })
```

Comment:
Using **RTK Query** for polling was super cool... Very comparable experience to **TanStack Query** I suppose.

I discovered that with RTK Query [we can also provide a custom `queryFn` function](https://redux-toolkit.js.org/rtk-query/usage/customizing-queries#customizing-queries-with-queryfn), just like with TanStack, where we can use our `client` to make the requests. The only problem I faced is on our side: We need to catch for exceptions and return the HTTP status code, but [web3 is not returning it](https://github.com/alephium/alephium-web3/blob/master/packages/web3/src/api/utils.ts#L22-L29).

Another thing I did not look for answers but was in my mind is: How do we ensure that the `client` object that is referenced inside the `queryFn`s is always up to date when the network settings change?

Reply:
I feel you. Because I started to created the Redux boilerplate, it indeed made sense going that way fully.
We may try to improve this once we'll implement fungible tokens historical balances.